tests/datastore_pymongo/test_pymongo.py

```diff
@@ -46,163 +50,244 @@ def _exercise_mongo(db):
-        ('Datastore/statement/MongoDB/%s/aggregate' % MONGODB_COLLECTION, 1),
-        ('Datastore/statement/MongoDB/%s/aggregate_raw_batches' % MONGODB_COLLECTION, 1),
-        ('Datastore/statement/MongoDB/%s/find_one_and_delete' % MONGODB_COLLECTION, 1),
-        ('Datastore/statement/MongoDB/%s/find_one_and_replace' % MONGODB_COLLECTION, 1),
-        ('Datastore/statement/MongoDB/%s/find_one_and_update' % MONGODB_COLLECTION, 1),
-    ])
-
-# Add Connection metric
-
-_test_pymongo_connection_scoped_metrics = (_test_pymongo_scoped_metrics +
-        [('Function/pymongo.connection:Connection.__init__', 1)])
-
-_test_pymongo_connection_rollup_metrics = (_test_pymongo_rollup_metrics +
-        [('Function/pymongo.connection:Connection.__init__', 1)])
-
-
-@pytest.mark.skipif(pymongo.version_tuple >= (3, 0),
-        reason='PyMongo version does not have pymongo.Connection.')
-@validate_transaction_errors(errors=[])
-@validate_transaction_metrics(
-        'test_pymongo:test_mongodb_connection_operation',
-        scoped_metrics=_test_pymongo_connection_scoped_metrics,
-        rollup_metrics=_test_pymongo_connection_rollup_metrics,
-        background_task=True)
-@background_task()
-def test_mongodb_connection_operation():
-    connection = pymongo.Connection(MONGODB_HOST, MONGODB_PORT)
-    db = connection.test
-    _exercise_mongo(db)
+    db[MONGODB_COLLECTION].create_index("x")
 
+    for item in db[MONGODB_COLLECTION].find().sort("x", pymongo.ASCENDING):
+        item["x"]
 
-# Add MongoClient metric
+    [item["x"] for item in db[MONGODB_COLLECTION].find().limit(2).skip(1)]
 
-_test_pymongo_mongo_client_scoped_metrics = (_test_pymongo_scoped_metrics +
-        [('Function/pymongo.mongo_client:MongoClient.__init__', 1)])
+    db[MONGODB_COLLECTION].bulk_write([pymongo.InsertOne({"x": 1})])
+    db[MONGODB_COLLECTION].insert_one({"x": 300})
+    db[MONGODB_COLLECTION].insert_many([{"x": 1} for i in range(20, 25)])
+    db[MONGODB_COLLECTION].replace_one({"x": 1}, {"x": 2})
+    db[MONGODB_COLLECTION].update_one({"x": 1}, {"$inc": {"x": 3}})
+    db[MONGODB_COLLECTION].update_many({"x": 1}, {"$inc": {"x": 3}})
+    db[MONGODB_COLLECTION].delete_one({"x": 4})
+    db[MONGODB_COLLECTION].delete_many({"x": 4})
+    db[MONGODB_COLLECTION].find_raw_batches()
+    db[MONGODB_COLLECTION].create_indexes([pymongo.IndexModel([("x", pymongo.DESCENDING)])])
+    db[MONGODB_COLLECTION].list_indexes()
+    db[MONGODB_COLLECTION].aggregate([])
+    db[MONGODB_COLLECTION].aggregate_raw_batches([])
+    db[MONGODB_COLLECTION].find_one_and_delete({"x": 10})
+    db[MONGODB_COLLECTION].find_one_and_replace({"x": 300}, {"x": 301})
+    db[MONGODB_COLLECTION].find_one_and_update({"x": 301}, {"$inc": {"x": 300}})
 
-_test_pymongo_mongo_client_rollup_metrics = (_test_pymongo_rollup_metrics +
-        [('Function/pymongo.mongo_client:MongoClient.__init__', 1)])
 
+def _exercise_mongo(db):
```

Comment:
It's a little confusing to have the switch on pymongo version in two places. I think it would be easier to follow and cleaner if this logic was moved into `_test` on ~ line 285:
```
    def _test():
        client = pymongo.MongoClient(MONGODB_HOST, MONGODB_PORT)
        db = client.test
        if pymongo.version_tuple < (4, 0):
            _exercise_mongo_v3(db)
        else:
            _exercise_mongo_v4(db)
```

Reply:
The problem would be that `_exercise_mongo` is used in all the test functions, not just that client test. It's necessary for the other tests to work and if we got rid of the top level one then we would need to duplicate the logic across all tests. I'd rather keep it where it is.